src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -114,13 +114,13 @@ public Person toModelType() throws IllegalValueException {
         if (!Address.isValidAddress(address)) {
             throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
         }
-        if (!Matric.isValidMatric(matric)) {
+        if (!Matric.isValidConstructorParam(matric)) {
             throw new IllegalValueException(Matric.MESSAGE_CONSTRAINTS);
         }
-        if (!Reflection.isValidReflection(reflection)) {
+        if (!Reflection.isValidConstructorParam(reflection)) {
             throw new IllegalValueException(Reflection.MESSAGE_CONSTRAINTS);
         }
-        if (!Studio.isValidStudio(studio)) {
+        if (!Studio.isValidConstructorParam(studio)) {
```

Comment:
Nice catch, this massive bug would have prevented our users from using this application properly. Lets continue to keep our application bug free!